### PR TITLE
Add datasets tests for FDS

### PR DESCRIPTION
**Pull request by adam-narozniak**
## Issue
Many of the datasets are not tested.

### Description

The test would enable checking the integration of the datasets with the FDS library.

## Proposal

Mocking datasets to enable the tests of bigger datasets.

### Explanation

* New file `mock_utils.py` contains 1) the code for mocking specific columns 2) the code for mocking specific datasets
* Change and extend the `federated_datasets_test.py`

### Changelog entry
<skip>

### CODE REVIEW

File: datasets/flwr_datasets/federated_dataset_test.py [hunk 1]
```diff
@@ -20,29 +20,70 @@
...
+        # Audio
+        ("speech_commands", "test", "v0.01"),
+    ],
 )
-class RealDatasetsFederatedDatasetsTrainTest(unittest.TestCase):
-    """Test Real Dataset (MNIST, CIFAR10) in FederatedDatasets."""
+class BaseFederatedDatasetsTest(unittest.TestCase):
+    """Test Real/Mocked Datasets used in FederatedDatasets.
+
+    The setUp method mocks the dataset download via datasets.load_dataset  if it is in
```
Comment (jafermarq) [inline datasets/flwr_datasets/federated_dataset_test.py:62]:
```suggestion
    The setUp method mocks the dataset download via datasets.load_dataset if it is in
```

File: datasets/flwr_datasets/mock_utils.py [hunk 2]
```diff
@@ -0,0 +1,377 @@
+# Copyright 2023 Flower Labs GmbH. All Rights Reserved.
```
Comment (jafermarq) [inline datasets/flwr_datasets/mock_utils.py:1]:
```suggestion
# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
```

**Review by jafermarq — DISMISSED** [review]

**Review by danieljanes — COMMENTED** [review]
The name of `mock_utils` needs to be changed, otherwise the build system will include it in the final release. Only files ending in `_test` are excluded.

**Comment by adam-narozniak** [comment]
I changed the name from `mock_utils` to `mock_utils_test`

**Review by danieljanes — APPROVED** [review]

**Pull request merged**